packages/migrations/src/migrateDockerNetworkIpRange/ensureDockerNetworkConfig/ensureDockerNetworkConfig.ts

```diff
@@ -0,0 +1,87 @@
+import { docker } from "@dappnode/dockerapi";
+import { logs } from "@dappnode/logger";
+import Dockerode from "dockerode";
+import { isNetworkOverlappingSubnet } from "./isNetworkOverlappingSubnet.js";
+import { recreateDockerNetwork } from "./recreateDockerNetwork.js";
+import { removeNetworksOverlappingSubnetIfNeeded } from "./removeNetworksOverlappingSubnetIfNeeded.js";
+
+/**
+ * Ensures a docker network configuration for a given parameters:
+ *
+ * - docker network "networkName" exists and it has assigned the subnet "networkSubnet"
+ * - All the previous docker containers mut be still connected to it at the end with the same
+ * aliases
+ *
+ * This function takes into account the following edge cases:
+ * - 404 network not found
+ * - 403 other network overlapping the address
+ *
+ * @param networkName "dncore_network"
+ * @param networkSubnet "172.33.0.1/16"
+ */
+export async function ensureDockerNetworkConfig({
+  networkName,
+  networkSubnet,
+}: {
+  networkName: string;
+  networkSubnet: string;
+}): Promise<{ network: Dockerode.Network; isNetworkRecreated: boolean }> {
+  const networkOptions: Dockerode.NetworkCreateOptions = {
+    Name: networkName,
+    Driver: "bridge",
+    IPAM: {
+      Driver: "default",
+      Config: [
+        {
+          Subnet: networkSubnet,
+        },
+      ],
+    },
+  };
+
+  try {
+    const dncoreNetwork = docker.getNetwork(networkName);
+    // docker network inspect
+    // https://docs.docker.com/engine/api/v1.43/#tag/Network/operation/NetworkInspect
+    const networkInspect: Dockerode.NetworkInspectInfo =
+      await dncoreNetwork.inspect(); // throws 404 if network not found
+
+    logs.info(`docker network ${networkName} exists`);
+
+    if (isNetworkOverlappingSubnet(networkInspect, networkSubnet)) {
+      // network subnet is as expected
+      logs.info(
+        `docker network ${networkName} has correct subnet ${networkSubnet}`
+      );
+      return { network: dncoreNetwork, isNetworkRecreated: false };
+    } else {
+      logs.warn(
+        `docker network ${networkName} has incorrect subnet ${networkInspect.IPAM?.Config?.[0].Subnet}, it should be ${networkSubnet}. Recreating it...`
+      );
+      await removeNetworksOverlappingSubnetIfNeeded(networkSubnet).catch((e) =>
+        logs.error(`error removing overlapping networks: ${e}`)
+      );
+      // CRITICAL: if this step fails migration failure
+      const network = await recreateDockerNetwork(
+        dncoreNetwork,
+        networkOptions
+      );
+      return { network, isNetworkRecreated: true };
+    }
+  } catch (e) {
+    if (e.statusCode === 404) {
+      // docker network not found, create it
+      logs.warn(`docker network ${networkName} not found, creating it...`);
+
+      await removeNetworksOverlappingSubnetIfNeeded(networkSubnet).catch((e) =>
```

Comment:
We cannot omit this error, it will not let us create the network

Reply:
But it worths trying to continue with the flow instead of blocking it.

Also that function does `listNetworks()` which could fail without meaning there are overlapping networks

Reply:
But we should try to "fix" the errors that might be thrown there. We might need to find any edge cases that could happen there and act accordingly

Reply:
As I said the `docker.listNetworks()` might fail, and that would prevent the flow of creating the network while it could be allowed to be created.

Reply:
It could fail, but I'd say it is extremely weird that listing the networks fails.... And that would not be a reason why not to try to handle the errors that might be thrown (or just retry), even if you end up trying to create the network anyway

Reply:
still, it worth moving forward. Ofcourse, taking into account as many edge cases as possible